transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -438,6 +440,320 @@ func ConvertExtendedSpatialIDsToQuadkeysAndAltitudekeys(extendedSpatialIDs []str
+//
+// 引数 :
+//
+//	request : 変換対象のTileXYZ構造体のスライス
+//	zBaseExponent： zの高さが1mとなるズームレベル
+//	zBaseOffset： ズームレベルがzBaseExponentのとき高度0mにおけるvZoom
+//	extendedSpatialIdVZoom : 拡張空間ID変換中の拡張空間IDの高さの精度指定。空間ID変換時、水平精度の方が高い場合この値は使われない。拡張空間IDの精度の閾値である 0 ～ 35 の整数値を指定可能。
+//
+// 戻り値 :
+//
+//	変換後の空間IDのスライス
+//
+// 戻り値(エラー) :
+//
+//	以下の条件に当てはまる場合、エラーインスタンスが返却される。
+//	 精度閾値超過(出力精度)：出力の水平方向精度、または高さ方向の精度に 0 ～ 35 の整数値以外が入力されていた場合。
+//	 z高度範囲外：変換前のzがその垂直ズームレベルにおける高度範囲外である場合。
+//	 拡張空間ID高度範囲外：拡張空間ID変換時の拡張空間ID高度がその垂直ズームレベルにおける高度範囲外である場合。
+//
+// 補足事項：
+//
+//	入力のoutputVZoomは次の2項目と関係しており、出力空間ID数に影響を与える
+//	1. TileXYZの垂直精度, zBaseExponent, zBaseOffset: 条件により拡張空間ID変換後の拡張空間ID数が増加する(詳しくは ConvertTileXYZsToExtendedSpatialIDs の例を参照)
+//	2. TileXYZの水平精度の差: 差が1増えるごとに1で出力された拡張空間ID数の4のべき乗で出力空間ID数が増加する。
+//	そのため、これら2項目の値によっては変換後の空間ID数は大幅に増大する。
+//	動作環境によってはメモリ不足となる可能性があるため、注意すること。
+func ConvertTileXYZsToSpatialIDs(request []*object.TileXYZ, zBaseExponent uint16, zBaseOffset int64, extendedSpatialIdVZoom uint16) ([]string, error) {
+	var outputData []string
+	outputExtendedSpatialIds, err := ConvertTileXYZsToExtendedSpatialIDs(
+		request, zBaseExponent, zBaseOffset, extendedSpatialIdVZoom,
+	)
+	if err != nil {
+		return nil, err
+	}
+	for _, out := range outputExtendedSpatialIds {
+		spatialIds := ConvertExtendedSpatialIDToSpatialIDs(&out)
+		outputData = append(outputData, spatialIds...)
+	}
+	return outputData, nil
+}
+
+// ConvertAltitudeKeyToZ altitudekeyを(拡張)空間IDのz成分に高度変換する。
+//
+// 変換前と変換後の精度差によって出力されるaltitudekeyの個数は増減する。
+//
+// 引数 :
+//
+//	altitudekey : 変換前のaltitudekeyの値
+//
+//	altitudekeyZoom : 変換前のaltitudekeyの精度指定
+//
+//	outputZoom : 変換対象の拡張空間ID垂直精度の指定
+//
+//	zBaseExponent : 変換対象の拡張空間IDの高さが1mとなるズームレベル
+//
+//	zBaseOffset : ズームレベルがzBaseExponentのとき、高度0mにおける拡張空間IDの垂直インデックス値
+//
+// 戻り値 :
+//
+//	変換後のaltitudekeyのスライス
+//
+// 戻り値(エラー) :
+//
+//	以下の条件に当てはまる場合、エラーインスタンスが返却される。
+//	 入力インデックス不正       ：inputIndexにそのズームレベル(inputZoom)で存在しないインデックス値が入力されていた場合。
+//	 出力インデックス不正       ：出力altitudekeyが出力ズームレベル(outputZoom)で存在しないインデックス値になった場合。
+func ConvertAltitudeKeyToZ(altitudekey int64, altitudekeyZoomLevel int64, outputZoom int64, zBaseExponent int64, zBaseOffset int64) (int64, int64, error) {
+	// 1. check that the input index exists in the input system
+	inputResolution := common.CalculateArithmeticShift(1, altitudekeyZoomLevel)
+
+	maxInputIndex := inputResolution - 1
+	minInputIndex := int64(0)
+
+	if altitudekey > maxInputIndex || altitudekey < minInputIndex {
+		return 0, 0, errors.NewSpatialIdError(errors.InputValueErrorCode, "input index does not exist")
+	}
+
+	// 2. Calculate internal index
+	zoomDifference := zBaseExponent - altitudekeyZoomLevel
+
+	internalMinIndex := common.CalculateArithmeticShift(altitudekey, zoomDifference)
+	internalMaxIndex := internalMinIndex
+	if zoomDifference > 0 {
+		internalMaxIndex = common.CalculateArithmeticShift(altitudekey+1, zoomDifference) - 1
+	}
+	// 3. Calculate outputMinIndex
+	outputZoomDifference := outputZoom - 25
```

Comment:
Use `zOriginValue`.

Reply:
fixed in [1aec8a9](https://github.com/trajectoryjp/spatial_id_go/pull/29/commits/1aec8a920ba5fc8387c4440a310343e63f95f2f8)